test/conformance/conformance_flowgraph.h

```diff
@@ -59,32 +62,429 @@ struct first_functor {
+
+    O operator()(oneapi::tbb::flow::continue_msg){
+        return 1;
+    }
+
+    O operator()(O){
+        return 1;
+    }
+
+    void operator()( const I& argument, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
+       std::get<0>(op).try_put(argument);
+    }
+};
+
+template <typename OutputType = int>
+struct passthru_body {
+    OutputType operator()( const OutputType& i ) {
+        return i;
+    }
+
+    OutputType operator()( oneapi::tbb::flow::continue_msg ) {
+       return OutputType();
+    }
+
+    void operator()( const int& argument, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
+       std::get<0>(op).try_put(argument);
+    }
+    
+    void operator()( const CountingObject<int>& argument, oneapi::tbb::flow::multifunction_node<CountingObject<int>, std::tuple<CountingObject<int>>>::output_ports_type &op ) {
+       std::get<0>(op).try_put(argument);
+    }
+};
+
+template<typename Node, typename Body = conformance::counting_functor<int>, typename ...Args>
+void test_body_exec_impl(Args... args){
+    oneapi::tbb::flow::graph g;
+    Body cf;
+    cf.execute_count = 0;
+
+    Node node1(g, args..., cf);
+
+    const size_t n = 10;
+    for(size_t i = 0; i < n; ++i) {
+        CHECK_MESSAGE((node1.try_put(oneapi::tbb::flow::continue_msg()) == true),
+                      "try_put of first node should return true");
+    }
+    g.wait_for_all();
+
+    CHECK_MESSAGE( (cf.execute_count == n), "Body of the first node needs to be executed N times");
+}
+
+template<typename Node, typename Body>
+void test_body_copying(Node& tested_node, Body& base_body){
+    using namespace oneapi::tbb::flow;
+
+    Body b2 = copy_body<Body, Node>(tested_node);
+
+    CHECK_MESSAGE( (base_body.copy_count + 1 < b2.copy_count), "copy_body and constructor should copy bodies");
+}
+
+template<typename Node, typename Body, typename ...Args>
+void test_copy_body_impl(Args... args){
+    using namespace oneapi::tbb::flow;
+
+    Body base_body;
+
+    graph g;
+
+    Node tested_node(g, args..., base_body);
+
+    Body b2 = copy_body<Body, Node>(tested_node);
+
+    CHECK_MESSAGE( (base_body.copy_count + 1 < b2.copy_count), "copy_body and constructor should copy bodies");
+}
+
+template<typename Node, typename ...Args>
+void test_buffering_impl(Args... args){
+    oneapi::tbb::flow::graph g;
+    conformance::dummy_functor<int> fun;
+    Node tested_node(g, args..., fun);
+    oneapi::tbb::flow::limiter_node<int> rejecter(g, 0);
+
+    oneapi::tbb::flow::make_edge(tested_node, rejecter);
+    tested_node.try_put(oneapi::tbb::flow::continue_msg());
+
+    int tmp = -1;
+    CHECK_MESSAGE( (tested_node.try_get(tmp) == false), "try_get after rejection should not succeed");
+    CHECK_MESSAGE( (tmp == -1), "try_get after rejection should not alter passed value");
+    g.wait_for_all();
+}
+
+template<typename Node, typename ...Args>
+void test_forwarding_impl(Args... args){
```

Comment:
I think it makes sense to test forwarding with more successors numbers (1,2,...,10).